leapp/utils/lock.py

```diff
@@ -0,0 +1,98 @@
+import os
+import fcntl
+import logging
+
+from leapp.config import get_config
+from leapp.exceptions import ProcessLockError
+
+
+def leapp_lock(lockfile=None):
+    return ProcessLock(lockfile=lockfile)
+
+
+def _acquire_lock(fd):
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True
+    except OSError:
+        return False
+
+
+def _read_pid(fd):
+    return os.read(fd, 20)
+
+
+def _write_pid(fd, pid):
+    os.write(fd, str(pid).encode('utf-8'))
+
+
+def _validate_pid(old_pid):
+    try:
+        old_pid = int(old_pid)
+    except ValueError:
+        return False
+
+    return old_pid
+
+
+def _is_old_process_running(old_pid):
+    return os.access('/proc/{}/stat'.format(old_pid), os.F_OK)
+
+
+def _clear_lock(fd):
+    os.lseek(fd, 0, os.SEEK_SET)
+    os.ftruncate(fd, 0)
+
+
+class ProcessLock(object):
+
+    def __init__(self, lockfile=None):
+        self.log = logging.getLogger('leapp.utils.lock')
+        self.lockfile = lockfile if lockfile else get_config().get('lock', 'path')
+
+    def _try_lock(self, pid):
+        fd = os.open(self.lockfile, os.O_CREAT | os.O_RDWR, 0o644)
```

Comment:
```suggestion
        fd = os.open(self.lockfile, os.O_CREAT | os.O_RDWR, 0o600)
```

Security thing I just learned: This needs to be `0o600`.  The way `fcntl()` and `flock()` work, a read lock can be taken out on a file if the user can read the file.  A read lock will block a write lock.  So, if the file is readable by other users, the other users could create a denial of service against LEAPP by asking for a read lock on the file.